### PR TITLE
feature/altitude_point

### PR DESCRIPTION
**Pull request by teamagam1**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/97" target="_blank">ReviewNinja

File: app/src/main/assets/GimelGimel/CesiumEventHandler.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ GG.EventHandler.prototype.pickEntity = function (windowPosition) {
             return id;
         }
     }
+    ;
```
Comment (teamagam3) [inline app/src/main/assets/GimelGimel/CesiumEventHandler.js]: Move to the line above or remove it if you can.

File: app/src/main/assets/GimelGimel/CesiumEventHandler.js [hunk 2]
```diff
@@ -48,43 +49,60 @@ GG.EventHandler.prototype.setScreenSpaceEventAction = function (screenSpaceEvent
...
-
-        }
-    });
+GG.EventHandler.prototype.onSingleTap = function (relativeX, relativeY) {
+    var self = this;
+    var position = self.getWindowPosition(relativeX, relativeY);
+    var entity = this.pickEntity(position);
+    clickEntityWithLayer(entity);
 
     function clickEntityWithLayer(entity) {
```
Comment (teamagam3) [inline app/src/main/assets/GimelGimel/CesiumEventHandler.js]: I'm not sure if you did it, but function inside a function isn't good.

It should be split to a separate "standalone" function.

File: app/src/main/assets/GimelGimel/CesiumEventHandler.js [hunk 3]
```diff
@@ -48,43 +49,60 @@ GG.EventHandler.prototype.setScreenSpaceEventAction = function (screenSpaceEvent
...
             } catch (err) {
-//                    iterate to next layer
+//              iterate to next layer
             }
         });
     }
 };
 
+GG.EventHandler.prototype.onLongPress = function (relativeX, relativeY) {
+    var self = this;
```
Comment (teamagam3) [inline app/src/main/assets/GimelGimel/CesiumEventHandler.js:59]: There is no need to declare this variable, you should do it if you need "this" inside inner / async method.

File: app/src/main/assets/GimelGimel/CesiumEventHandler.js [hunk 4]
```diff
@@ -48,43 +49,60 @@ GG.EventHandler.prototype.setScreenSpaceEventAction = function (screenSpaceEvent
...
+        longitude: Cesium.Math.toDegrees(cartographic.longitude),
+        latitude: Cesium.Math.toDegrees(cartographic.latitude),
+        altitude: cartographic.height,
+        hasAltitude: true
+    });
+
+};
+
+GG.EventHandler.prototype.onDoubleTap = function (relativeX, relativeY) {
+    var self = this;
```
Comment (teamagam3) [inline app/src/main/assets/GimelGimel/CesiumEventHandler.js:75]: Same same

File: app/src/main/assets/GimelGimel/CesiumEventHandler.js [hunk 5]
```diff
@@ -48,43 +49,60 @@ GG.EventHandler.prototype.setScreenSpaceEventAction = function (screenSpaceEvent
...
-            GG.AndroidAPI.updateSelectedLocation({
-                            longitude: longitude,
-                            latitude: latitude
-                        });
-
-
-        }
-    });
+GG.EventHandler.prototype.onSingleTap = function (relativeX, relativeY) {
+    var self = this;
```
Comment (teamagam3) [inline app/src/main/assets/GimelGimel/CesiumEventHandler.js]: You can fix it here too

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/viewer_footer_fragments/MapManipulationFooterFragment.java [hunk 6]
```diff
@@ -49,7 +49,8 @@ public void onClick(View v) {
     }
 
     private void displayLastTouchedLocation() {
-        PointGeometry pg = getInterface().getGGMap().getLastTouchedLocation();
+//        PointGeometry pg = getInterface().getGGMap().getLastTouchedLocation();
+        PointGeometry pg = new PointGeometry(1, 2);
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/viewer_footer_fragments/MapManipulationFooterFragment.java]: What is the purpose of this object?

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java [hunk 7]
```diff
@@ -399,4 +387,32 @@ private void validateLayerExists(String layerId) {
             throw new IllegalArgumentException("layer not found with Id: " + layerId);
         }
     }
+
+    private class CesiumUIJavascriptCommandExecutor implements CesiumBaseBridge.JavascriptCommandExecutor {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: Separate class file would be cleaner.

You can pass the necessary variable in the constructor.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/JavascriptInterfaces/CesiumMapGestureDetector.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @Override
+    public boolean onDoubleTap(MotionEvent event) {
+        mCesiumGestureBridge.onDoubleTap(event.getX() / mViewWidth, event.getY() / mViewHeight);
+        return true;
+    }
+
+    @Override
+    public void onLongPress(MotionEvent event) {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/JavascriptInterfaces/CesiumMapGestureDetector.java]: Why the other methods return boolean and this one returns void?

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/JavascriptInterfaces/CesiumMapGestureDetector.java [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+    /***
+     * A method exposed for javascript execution.
+     * Updates selected location
+     *
+     * @param locationJson - An json with latitude and longitude
+     *                     properties describing current location
+     */
+    @JavascriptInterface
+    public void onLongPress(String locationJson) {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/JavascriptInterfaces/CesiumMapGestureDetector.java]: Change to a better name.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/JavascriptInterfaces/CesiumMapGestureDetector.java [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+        PointGeometry pg = CesiumUtils.getPointGeometryFromJson(locationJson);
+        mOnMapGestureListener.onLongPress(pg);
+    }
+
+
+    @JavascriptInterface
+    public void onDoubleTap(String locationJson) {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/JavascriptInterfaces/CesiumMapGestureDetector.java]: Same

File: app/src/main/assets/GimelGimel/CesiumEventHandler.js (hunk 1 above)
Comment (teamagam1) [inline app/src/main/assets/GimelGimel/CesiumEventHandler.js]: !fixed

File: app/src/main/assets/GimelGimel/CesiumEventHandler.js (hunk 2 above)
Comment (teamagam1) [inline app/src/main/assets/GimelGimel/CesiumEventHandler.js]: !resolved

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/viewer_footer_fragments/MapManipulationFooterFragment.java (hunk 6 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/viewer_footer_fragments/MapManipulationFooterFragment.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java (hunk 7 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/JavascriptInterfaces/CesiumMapGestureDetector.java (hunk 9 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/JavascriptInterfaces/CesiumMapGestureDetector.java]: !fixed

**Pull request merged**